src/python/WMCore/Services/PyCondor/PyCondorAPI.py

```diff
@@ -18,34 +17,64 @@
 
 
 class PyCondorAPI(object):
+    """
+    Some APIs to interact with HTCondor via the HTCondor python bindings.
+    """
     def __init__(self):
         self.schedd = htcondor.Schedd()
         self.coll = htcondor.Collector()
 
-    def getCondorJobs(self, constraint, attr_list):
+    def recreateSchedd(self):
+        """
+        In case our current schedd object is in a "strange" state, we
+        better recreate it.
         """
-        _getCondorJobs_
+        logging.warning("Recreating Schedd instance due to query error...")
+        self.schedd = htcondor.Schedd()
 
-        Given a job/schedd constraint, return a list of jobs attributes
-        or None if the query to condor fails.
+    def getCondorJobsSummary(self):
+        """
+        Retrieves a job summary from the HTCondor Schedd object
+        :return: a list of classads representing the matching jobs, or None if failed
         """
+        ### NOTE: SummaryOnly does not work with xquery method
```

Comment:
I know in this case it doesn't matter if the background query is `query` or `xquery` because we do expect only the total numbers to be provided  by HTcondor, but just for my curiosity, is it documented in the HTCondor the fact that this option is not supported with `xquery` or you had to discover it yourself?

Reply:
I haven't seen any documentation on that, so I had to find it the hard way.
If it isn't too expensive to file an HTCondor issue, I might do so (even though we are not using their latest version...)